### PR TITLE
Feature: Azure + Azure-SSIS Integration Runtimes for Azure Data Factory

### PR DESCRIPTION
**Pull request by fernandoBRS**
## Problem

In Azure Data Factory, it is possible to create 3 types of Integration Runtimes:

- Azure
- Azure-SSIS
- Self-hosted

However, currently Terraform supports only 2 types:

- **Azure-SSIS** IR through the `azurerm_data_factory_integration_runtime_managed` resource.
- **Self-hosted** IR through the `azurerm_data_factory_integration_runtime_self_hosted` resource.

The purpose of this PR is to add support for **Azure** Integration Runtime.

## Proposal

In order to avoid confusion on describing **Azure** and **Azure-SSIS** IR resources, the following approach was followed:

- The `azurerm_data_factory_integration_runtime_azure` resource was created to allow the definition of Azure Integration Runtime.
- The `azurerm_data_factory_integration_runtime_azure_ssis` resource was created to allow the definition of Azure-SSIS Integration Runtime.
- The existing `azurerm_data_factory_integration_runtime_managed` was marked as **deprecated** in favor of `azurerm_data_factory_integration_runtime_azure_ssis`.

> An important aspect to consider is that Azure and Azure-SSIS IRs have different interfaces for compute properties, making even more sense to split them into two resources.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Hi @fernandoBRS - thank you for this PR, however this would be a breaking change for users of the existing resource. Instead of changing the behaviour of it could we perhapes deprecate it and copy its functionality to `azurerm_data_factory_integration_runtime_azure_ssis` and create a new one `azurerm_data_factory_integration_runtime_azure` ?

**Comment by fernandoBRS** [comment]
Great suggestions @katbyte! I'll be working on these changes.

**Comment by fernandoBRS** [comment]
@katbyte I made the suggested changes. Now we have two new resources:

- `azurerm_data_factory_integration_runtime_azure` for Azure Integration Runtime
- `azurerm_data_factory_integration_runtime_azure_ssis` for Azure-SSIS Integration Runtime

The `azurerm_data_factory_integration_runtime_managed` was marked as deprecated in the doc, with recommendations for `azurerm_data_factory_integration_runtime_azure_ssis` usage for new deployments.

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @fernandoBRS - aside from a few comments this looks good. Once they are addressed this should be good to merge!

Also we'll need to update with master to fix the build:
```
[Step 4/5] ./data_factory_integration_runtime_azure_resource.go:56:19: undefined: "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate".DataFactoryName
[02:58:23]	[Step 4/5] ./data_factory_integration_runtime_azure_ssis_resource.go:56:19: undefined: "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate".DataFactoryName
[02:58:23]	[Step 4/5] Process exited with code 2
```

File: azurerm/internal/services/datafactory/data_factory_integration_runtime_azure_resource.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+			"core_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  8,
+				ValidateFunc: validation.IntInSlice([]int{
+					8, 16, 32, 48, 80, 144, 272,
+				}),
+			},
+
+			"time_to_live": {
```
[the remark this answers is not in the document]
Comment (fernandoBRS) [reply azurerm/internal/services/datafactory/data_factory_integration_runtime_azure_resource.go]: Sure, I changed it and updated the documentation as well to reflect the property name.

File: azurerm/internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+				check.That(data.ResourceName).Key("catalog_info.0.administrator_login").HasValue("ssis_catalog_admin"),
+				check.That(data.ResourceName).Key("catalog_info.0.administrator_password").HasValue("my-s3cret-p4ssword!"),
+				check.That(data.ResourceName).Key("catalog_info.0.pricing_tier").HasValue("Basic"),
+			),
+		},
+		data.ImportStep("catalog_info.0.administrator_password"),
+	})
+}
+
+func TestAccDataFactoryIntegrationRuntimeManagedSsis_customSetupScript(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (fernandoBRS) [reply azurerm/internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource_test.go]: Yes, I created a `TestAccDataFactoryIntegrationRuntimeManagedSsis_complete` function that combines the tests for VNet integration, custom scripts and catalog info. Way easier, thanks for the suggestion!

File: azurerm/internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type IntegrationRuntimeManagedSsisResource struct {
+}
+
+func TestAccDataFactoryIntegrationRuntimeManagedSsis_basic(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (fernandoBRS) [reply azurerm/internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource_test.go:20]: I've updated the `basic` test to validate all required properties and created  `complete` test.

**Review by katbyte — APPROVED** [review]
Thanks @fernandoBRS - LGTM 👍

**Pull request merged**